core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -32,25 +44,48 @@ void shouldBeExecutedAfterLasCommitDateProbe() {
     }
 
     @Test
-    void shouldCheckIfTheUnreleasedCommitsExist() throws IOException {
+    void shouldCheckIfTheUnreleasedCommitsExist() throws IOException, GitAPIException {
         final Plugin plugin = mock(Plugin.class);
         final ProbeContext ctx = mock(ProbeContext.class);
         final HasUnreleasedProductionChangesProbe probe = getSpy();
 
-        when(plugin.getScm()).thenReturn("https://github.com/jenkinsci/git-plugin/tree/master/src");
-        when(ctx.getScmRepository()).thenReturn(Files.createTempDirectory(UUID.randomUUID().toString()));
-        final ProbeResult probeResult = probe.apply(plugin, ctx);
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
 
-        assertThat(probeResult.id()).isEqualTo("unreleased-production-changes");
-        assertThat(probeResult.status()).isEqualTo(ResultStatus.SUCCESS);
+//        ref: https://www.vogella.com/tutorials/JGit/article.html
+//        creating a fake Git repository
 
-        when(plugin.getScm()).thenReturn("https://github.com/jenkinsci/git-plugin/blob/master/pom.xml");
-        when(ctx.getScmRepository()).thenReturn(Files.createTempDirectory(UUID.randomUUID().toString()));
-        final ProbeResult probeResult2 = probe.apply(plugin, ctx);
+        Path parentDir = Paths.get("/path/to/local");
+        Path tempDir = Files.createTempDirectory(parentDir, "mytempdir");
 
-        assertThat(probeResult2.id()).isEqualTo("unreleased-production-changes");
-        assertThat(probeResult2.status()).isEqualTo(ResultStatus.SUCCESS);
+        Path srcTempDir = Files.createTempDirectory(tempDir, "filedir");
+//        create a commit
+//        https://stackoverflow.com/a/51151158
+        Git git = Git.init().setDirectory(tempDir.toFile()).call();
 
+        // create files to add to the repo
+
+        File file1 = new File(tempDir.toString(), "pom.xml");
+        File file2 = new File(tempDir.toString(), "README.md");
+        File file3 = new File(tempDir.toString(), String.valueOf(srcTempDir));
+        file1.createNewFile();
+        file2.createNewFile();
+        file3.createNewFile();
+
+        AddCommand add = git.add();
+        // Stage all files in the repo
+        add.addFilepattern("pom.xml").addFilepattern("README.md").addFilepattern("src/test/resources/file.txt").call();
+        Repository repository = new FileRepositoryBuilder().setGitDir(tempDir.toFile()).build();
+
+        // create committer
+        PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+        PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().plusDays(1).toInstant()));
+
+        // commit the files
+        CommitCommand commit = git.commit();
+        commit.setOnly( "pom.xml" ).setOnly( "README.md" ).setOnly("src/test/resources/file.txt").
+            setCommitter(committer).
+            setMessage("initial commit").call();
```

Comment:
Once this error gets resolved, I have to assert them? 

Do I need to create more data?

Reply:
technically, you are creating only one commit here, containing multiple files. This is again not really validating what you are trying to prove.

The probes needs to:

 - returns a failure when the last commit date of any file in `src/main` folder is more recent than the plugin release timestamp
 - returns a failure when the last commit date of `pom.xml` file is more recent than the plugin release timestamp
 - returns a success when the plugin release timestamp is more recent than the last commit date of `pom.xml` or any file in `src/main`